docs/design/ocs-to-ocs-api-server.md

```diff
@@ -0,0 +1,201 @@
+message StorageConfigResponse{
+    // data contains the json blob to be used by the consumer cluster to connect with the provider cluster
+   bytes data = 1;
+}
+
+```
+##### Response Scheme
+
+
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful. StorageConsumer CR is `Ready`, rook resources are created and external python script was executed successfully to generate the json connection blob.    | N/A |
+| Awaiting Ceph resources    | 14 UNAVAILABLE     | Waiting for the ceph resources to be created and updated in the StoragConsumer CR status    | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed due to invalid StorageConsumer ID | Contact the Provider Cluster Admin to verify the StorageConsumer ID
+| Error | 13 INTERNAL | Request failed due to some error in provider cluster | Contact Provider Cluster Admin
+
+#### OffboardConsumer
+- Make a delete request on the StorageConsumer CR and return OK.
+
+```protobuf
+// OffBoardConsumerRequest holds the required information to delete the StorageConsumer CR on the storage provider cluster
+message OffBoardConsumerRequest{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+}
+
+// OffBoardConsumerResponse holds the response for the OffBoardConsumer API request
+message OffBoardConsumerResponse{
+
+}
+
+```
+
+##### Response Scheme
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful     | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed due to invalid StorageConsumer ID | Contact the Provider Cluster Admin to verify the ID
+| Error | 13 INTERNAL | Request failed due to some error in provider cluster | Contact Provider Cluster Admin
+
+
+
+#### UpdateCapacity
+- Read the `status.state` of the StorageConsumer CR:
+  - if Status is `Ready`, then update the `spec.Capacity` and return OK
+
+```protobuf
+// UpdateCapacityRequest holds the information required to increase or decrease the block pool size
+// on the provider cluster
+message UpdateCapacityRequest{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+    // capacity is the desired storage requested by the consumer cluster
+    k8s.io.apimachinery.pkg.api.resource.Quantity capacity = 2;
+}
+
+// UpdateCapacityResponse holds the response for UpdateCapacity API request
+message UpdateCapacityResponse{
+
+}
+```
+
+##### Response Scheme
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful     | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed to invalid Consumer ID | Contact the Provider Cluster Admin to verify the StorageConsumer ID
+| Invalid Capacity | 3 INVALID_ARGUMENT | Request failed due to invalid capacity | Contact the Provider Cluster Admin to get the valid capacity value
```

Comment:
How do you distinguish between these two? Might it be more accurate to say:
```
Invalid | 3 INAVLID_ARGUMENT | Request failed due to invalid Consumer ID or Capacity | ....
```

Reply:
done